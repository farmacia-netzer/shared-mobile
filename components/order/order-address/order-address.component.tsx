import React from 'react';
import { StyleSheet, View } from 'react-native';

import { NetzerKeyValueText } from '##component/netzer-key-value-text/netzer-key-value-text.component';
import { TAddress } from '#types';

interface OrderAddressProp {
  address: TAddress;
}

export const OrderAddress: React.FC<OrderAddressProp> = ({ address: { address, references } }: OrderAddressProp) => {
  return (
    <View style={styles.container}>
      <NetzerKeyValueText header="Direccion" value={address} />
      <NetzerKeyValueText header="Referencia" value={references || 'N/A'} />
    </View>
  );
};
const styles = StyleSheet.create({
  container: {}
});
