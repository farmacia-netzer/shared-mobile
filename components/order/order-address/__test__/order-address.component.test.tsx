import React from 'react';
import { render } from '@testing-library/react-native';

import { OrderAddress } from '../order-address.component';

describe('<OrderAddress />', () => {
  it('renders', () => {
    const { toJSON } = render(<OrderAddress address={{} as any} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
