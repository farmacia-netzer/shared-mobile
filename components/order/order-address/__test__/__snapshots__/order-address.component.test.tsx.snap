// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderAddress /> renders 1`] = `
<View
  style={Object {}}
>
  <View>
    <Text
      fontStyle={12}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 18,
          },
          undefined,
          Object {
            "fontSize": 14,
            "marginBottom": 5,
            "marginTop": 15,
          },
        ]
      }
    >
      Direccion
    </Text>
    <Text
      fontStyle={12}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 18,
          },
          undefined,
          Object {
            "fontSize": 20,
            "fontWeight": "400",
            "lineHeight": 24,
          },
        ]
      }
    >
      N/A
    </Text>
  </View>
  <View>
    <Text
      fontStyle={12}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 18,
          },
          undefined,
          Object {
            "fontSize": 14,
            "marginBottom": 5,
            "marginTop": 15,
          },
        ]
      }
    >
      Referencia
    </Text>
    <Text
      fontStyle={12}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 18,
          },
          undefined,
          Object {
            "fontSize": 20,
            "fontWeight": "400",
            "lineHeight": 24,
          },
        ]
      }
    >
      N/A
    </Text>
  </View>
</View>
`;
